src/auth/certificatesubjecttokensupplier.ts

```diff
@@ -0,0 +1,299 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {GaxiosOptions} from 'gaxios';
+import {SubjectTokenSupplier} from './identitypoolclient';
+import {AuthClient} from './authclient';
+import {getWellKnownCertificateConfigFileLocation, isValidFile} from '../util';
+import * as fs from 'fs';
+import {X509Certificate} from 'crypto';
+import * as https from 'https';
+
+// --- Constants ---
+export const CERTIFICATE_CONFIGURATION_ENV_VARIABLE =
+  'GOOGLE_API_CERTIFICATE_CONFIG';
+
+// --- Custom Errors ---
+
+/**
+ * Thrown when the certificate source cannot be located or accessed.
+ */
+export class CertificateSourceUnavailableError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = 'CertificateSourceUnavailableError';
+  }
+}
+
+/**
+ * Thrown for invalid configuration that is not related to file availability.
+ */
+export class InvalidConfigurationError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = 'InvalidConfigurationError';
+  }
+}
+
+// --- Interfaces ---
+
+/**
+ * Defines options for creating a {@link CertificateSubjectTokenSupplier}.
+ */
+export interface CertificateSubjectTokenSupplierOptions {
+  /**
+   * If true, uses the default well-known location for the certificate config.
+   * Either this or `certificateConfigLocation` must be provided.
+   */
+  useDefaultCertificateConfig?: boolean;
+  /**
+   * The file path to the certificate configuration JSON file.
+   * Required if `useDefaultCertificateConfig` is not true.
+   */
+  certificateConfigLocation?: string;
+  /**
+   * The file path to the trust chain (PEM format). If not provided,
+   * only the leaf certificate will be sent.
```

Comment:
I think it's best if we remove the "If not provided, only the leaf certificate will be sent." part and just note that this field is optional. How we are creating the subject token to send to STS is an implementation detail that I think should not be exposed here, as it'd make the reader more confused than informed.